scripts/inference.py

```diff
@@ -181,8 +228,25 @@ def infer(use_cache, do_sample):
 
 print("generating output", local_rank)
 do_sample = [False]
-use_cache = [
-    args.no_use_cache
-]  # True/False are identical with greedy iff `torch.use_deterministic_algorithms(True)`
-for sample, cache in itertools.product(do_sample, use_cache):
+cache_type = args.cache_type
+kv_cache_manager = None
+if cache_type == "paged":
+    from fms.utils.cache.paged import PagedKVCacheManager
+
+    use_cache = True
+    kv_cache_manager = PagedKVCacheManager(
+        model.config.nlayers,
+        model.config.nheads,
+        model.config.emb_dim,
+        tensor_parallel_size=dist.get_world_size() if args.distributed else 1,
+        dtype=torch.get_default_dtype(),
+        device=device,
+    )
+elif cache_type == "expandable":
+    use_cache = True
+else:
```

Comment:
should error on incorrect values

Reply:
added exception here if not one of the valid choices